go-controller/pkg/util/util.go

```diff
@@ -387,50 +384,27 @@ func IsUnknownActiveNetworkError(err error) bool {
 	return errors.As(err, &unknownActiveNetworkError)
 }
 
-// GetActiveNetworkForNamespace returns the NetInfo struct of the active network
-// for the given namespace based on the NADs present in that namespace.
-// active network here means the network managing this namespace and responsible for
-// plumbing all the entities for this namespace
-// this is:
-// 1) &DefaultNetInfo if there are no NADs in the namespace OR all NADs are Role: "primary"
-// 2) &NetConf{Name: "<secondary-network-name>"} if there is exactly ONE NAD with Role: "primary"
-// 3) Multiple primary network role NADs ActiveNetworkUnknown error
-// 4) error under all other conditions
-func GetActiveNetworkForNamespace(namespace string, nadLister nadlister.NetworkAttachmentDefinitionLister) (NetInfo, error) {
-	if nadLister == nil {
-		return &DefaultNetInfo{}, nil
-	}
-	if !IsNetworkSegmentationSupportEnabled() {
-		return &DefaultNetInfo{}, nil
-	}
-	namespaceNADs, err := nadLister.NetworkAttachmentDefinitions(namespace).List(labels.Everything())
-	if err != nil {
-		return nil, err
-	}
-	if len(namespaceNADs) == 0 {
-		return &DefaultNetInfo{}, nil
-	}
-	numberOfPrimaryNetworks := 0
-	var primaryNetwork NetInfo
-	for _, nad := range namespaceNADs {
-		netInfo, err := ParseNADInfo(nad)
-		if err != nil {
-			klog.Warningf("Skipping nad '%s/%s' as active network after failing parsing it with %v", nad.Namespace, nad.Name, err)
-			continue
-		}
+func NewUnknownActiveNetworkError(namespace string) *UnknownActiveNetworkError {
```

Comment:
Are we using this one?

Reply:
I guess not anymore after we guarantee the controller only renders a single primary network.

Reply:
removed